src/realm-settings/ExecutorForm.tsx

```diff
@@ -55,54 +59,123 @@ export default function ExecutorForm() {
+
+      const profile = profiles.profiles!.find(
+        (profile) => profile.name === profileName
+      );
+
+      const profileExecutor = profile?.executors!.find(
+        (executor) => executor.executor === executorName
+      );
+
+      if (profileExecutor) {
+        Object.entries(profileExecutor).map(([key, value]) => {
+          if (key === "configuration") {
+            setValue("config", value);
+          }
+        });
+      }
     },
     []
   );
 
-  const fldNameFormatter = (name: string) =>
-    name.toLowerCase().trim().split(/\s+/).join("-");
-
   const save = async () => {
     const formValues = form.getValues();
     const updatedProfiles = profiles.map((profile) => {
       if (profile.name !== profileName) {
         return profile;
       }
 
+      const profileExecutor = profile.executors!.find(
+        (executor) => executor.executor === executorName
+      );
+
       const executors = (profile.executors ?? []).concat({
         executor: formValues.executor,
-        configuration: formValues.configuration,
+        configuration: formValues.config,
       });
 
-      return {
-        ...profile,
-        executors,
-      };
+      let editedExecutorConfig = {};
+      if (editMode) {
+        editedExecutorConfig = Object.assign(
+          profileExecutor!.configuration,
+          formValues.config
+        );
+        profileExecutor!.configuration = editedExecutorConfig;
+      }
+
+      if (editMode) {
+        return {
+          ...profile,
+        };
+      } else {
+        return {
+          ...profile,
+          executors,
+        };
+      }
     });
     try {
       await adminClient.clientPolicies.createProfiles({
         profiles: updatedProfiles,
         globalProfiles: globalProfiles,
       });
-      addAlert(t("realm-settings:addExecutorSuccess"), AlertVariant.success);
+      addAlert(
+        editMode
+          ? t("realm-settings:updateExecutorSuccess")
+          : t("realm-settings:addExecutorSuccess"),
+        AlertVariant.success
+      );
+
       history.push(toClientProfile({ realm, profileName }));
     } catch (error) {
-      addError("realm-settings:addExecutorError", error);
+      addError(
+        editMode
+          ? "realm-settings:updateExecutorError"
+          : "realm-settings:addExecutorError",
+        error
+      );
     }
   };
 
+  const globalProfile = globalProfiles.find(
+    (globalProfile) => globalProfile.name === profileName
+  );
+
+  const profileExecutorType = executorTypes?.find(
+    (executor) => executor.id === executorName
+  );
+
+  const editedProfileExecutors = profileExecutorType?.properties.map(
+    (property) => {
+      const globalDefaultValues = editMode ? property.defaultValue : "";
+
+      return {
+        helpText: property.helpText!,
+        label: property.label!,
+        name: property.name!,
+        defaultValue: globalDefaultValues,
+        type: property.type!,
+      };
```

Comment:
If you are copying the properties over perhaps you can use a spread here instead?
```suggestion
      return {
        ...property,
        defaultValue: globalDefaultValues,
      };
```

Reply:
ah, yeah, thanks!